### ISSUE TITLE
Invalid GraphicsMagick result

### ISSUE THREAD

**Issue by Neo-Oli** [Deathspike/mangarack#69]
Hi, I'm currently getting this error:
```
~/drive/comics $ mangarack-cli "http://mangafox.me/manga/one_piece/"
mangarack-cli (4.1.2)
Fetching One Piece
Fetching One Piece #866.cbz
Error: Invalid GraphicsMagick result. Is GraphicsMagick installed?
    at Object.<anonymous> (/lib/node_modules/mangarack/dist/..\mangarack-runnable-cli\src\services\imageService.ts:28:13)
    at step (/lib/node_modules/mangarack/mangarack-component-core/node_modules/tslib/tslib.js:122:1)
    at Object.throw (/lib/node_modules/mangarack/mangarack-component-core/node_modules/tslib/tslib.js:103:1)
    at rejected (/lib/node_modules/mangarack/mangarack-component-core/node_modules/tslib/tslib.js:94:1)
    at process._tickCallback (internal/process/next_tick.js:109:7)
~/drive/comics 1 $
```

It only occurs on chapter #866 of One Piece. All previous chapters downloaded without a problem. Graphicsmagick is definetly  Is there any way to debug this, see what the problem is?

**Comment by Deathspike** [comment]
Can you check on the website to see if there is an image that doesn't load, or partially loads?
If so, that would be the culprit. Then we can decide what to do.
(I would check, but I am currently unavailable)

**Comment by Neo-Oli** [comment]
Ah, you're right. It seems to be that #866 is the very latest chapter (I only recently started reading it, so I am only up to chapter #247). The very last image seems to be broken. `http://mangafox.me/manga/one_piece/vTBD/c866/21.html. It points to http://h.mfcdn.net/store/manga/106/TBD-866.0/compressed?token=d196ce921f9c5483fbbfd2e4513a934b&ttl=1496372400`, which seems to be a broken URL (missing file name). My guess is that there is not supposed to be a Page 21.

**Comment by Deathspike** [comment]
Right, so `mangarack` does what it is supposed to, but it could produce a better error :)

You probably want to report this broken image on the site so it can be fixed.

For now, I will not consider an option to skip broken images. Unless this occurs more frequently.

(Note to self: Check how many images 866 *should* have had after fixing)

**Comment by Neo-Oli** [comment]
Thanks. I did send them a message now.

Not skipping broken images seems fair, as these images might get fixed but leave people with broken archives. 
Would you consider skipping the URL and continuing to the next one?
I am running the command like this (as is suggested in the README.md) as a cronjob every day:
`cd ~/drive/comics&&mangarack-cli < download-list`
`download-list` looks like this:
```
http://mangafox.me/manga/one_piece/
http://mangafox.me/manga/yotsubato/
```
So if an error occurs while downloading One Piece, Yotsubato! will never get updated. 

A workaround might be running this instead:
`cd ~/drive/comics&&while read m; do mangarack-cli "$m";done <download-list`
This starts mangarack-cli for every URL separately.

**Comment by Deathspike** [comment]
That's a good thought 👍 I'll make an issue describing a change of behavior to implement this.

**Comment by Deathspike** [comment]
This is now available in 4.2.0. See #70. Do let me know if this works out for you 👍

**Comment by Neo-Oli** [comment]
Thanks. That seems to work now. Unfortunately One Piece from Mangafox still has that issue. It will always fail to download the very latest Chapter (it's 868 that fails now)

**Comment by XellossMetallium** [comment]
I had a similar problem a couple of times. One of these was "Hare kon" on batoto, chapter 90, page 3.
http://bato.to/reader#a4f1e93481d1069b_3
The new behavior (4.2.0, before I had excluded Hare kon link to avoid issues) is the following:
Hare Kon ch. 90 downloaded, only the first two pages (btw from 4 onward should be fine)
Hare Kon ch 91 skipped
Next link correctly processed.

I don't know if this is the best approach: Hare Kon ch. 90 is useless, yet the cbz is created and added to library. Ch 91 is skipped, and that on the long run could pose a problem: let's say that of 100 chapters only chapter 20 is broken... if no one fixed it before it could remain like this forever, so one could decide to read just that chapter online and switch back to cbz for the others. 
I suppose that it would be better to skip the broken chapter (or somehow at least warn the user that the chapter has missing images, something visible on comicrack and in windows explorer) but to continue downloading the following chapters.

Sorry if I missed something, I didn't follow this issue from the beginning!

**Comment by Neo-Oli** [comment]
I don't think it should create a .cbz if there was an error while downloading. I'd rather have a chapter missing than a broken one that never gets updated.

**Comment by Deathspike** [comment]
Oh, dear, what an oversight. That was not the intended behavior at all. I'll fix this in the evening!

**Comment by Deathspike** [comment]
Fixed in 4.2.1 (https://github.com/Deathspike/mangarack/commit/e24c8304f5eb54cfecfcb153602deb984b5cbd7e)
